internal/compiler/program.go

```diff
@@ -203,6 +205,14 @@ func NewProgram(opts ProgramOptions) *Program {
 
 	p.processedFiles = processAllProgramFiles(p.opts, libs, p.singleThreaded())
 
+	// Populate sourceFilesFoundSearchingNodeModules from the processed files
+	// This mirrors the TypeScript implementation's tracking behavior
+	if p.processedFiles.sourceFilesFoundSearchingNodeModules != nil {
+		for path, isExternal := range p.processedFiles.sourceFilesFoundSearchingNodeModules {
+			p.sourceFilesFoundSearchingNodeModules.Store(path, isExternal)
+		}
+	}
```

Comment:
This should not be required, the info should already be on processedFiles and therefore be accessible.